### PR TITLE
Update rustc-ux-guidelines

### PR DESCRIPTION
**Pull request by jz0425**
1) changed "long way into" to "long way toward"
2) changed "developer lives" to "developers' lives"
3) removed the "either... or..." format from second paragraph because there are more than 2 options
4) Minor revisions to paragraphs 3-6 to make them more consistent in format and to fix minor grammar issues.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Thanks for the contribution! I think the title of your PR was copy pasted incorrectly, though. Changes look good to me, but I'll leave to @frewsxcv for an official review.

File: src/doc/rustc-ux-guidelines.md [hunk 1]
```diff
@@ -17,11 +17,11 @@ An `error` is emitted when the compiler detects a problem that makes it unable
 A `warning` is emitted when the compiler detects something odd about a
 program. For instance, dead code and unused `Result` values.
 
-A `help` is emitted following either an `error` or `warning` giving extra
+A `help` message is emitted following an `error` or `warning` to give additional
 information to the user about how to solve their problem.
 
-A `note` is for identifying additional circumstances and parts of the code
-that lead to a warning or error. For example, the borrow checker will note any
+A `note` to emitted to identify additional circumstances and parts of the code
```
Comment (frewsxcv) [inline src/doc/rustc-ux-guidelines.md]: > A `note` to emitted to identify...

The wording here seems a little off and I'm having trouble understanding the intention. Maybe you mean:

> A `note` is emitted to identify...

Comment (jz0425) [inline src/doc/rustc-ux-guidelines.md]: Yes, that is what I meant, thanks for the catch.  Should I update and do a new PR?

**Comment by frewsxcv** [comment]
thanks for these improvements!

@bors r+ rollup

**Pull request merged**